cmd/object-handlers.go

```diff
@@ -1106,25 +1105,40 @@ func (api objectAPIHandlers) CopyObjectHandler(w http.ResponseWriter, r *http.Re
 				}
 			}
 
+			// FIXME: previously this was used to compute the
+			// encrypted payload size. It appears that we can
+			// compute more simply using ObjectInfo.EncrypedSize
+			// method if IsTargetEncrypted, which is the only case
+			// that matters here.
+			//
 			// Calculate the size of the target object
-			var targetSize int64
-
-			switch {
-			case isDstCompressed:
-				targetSize = -1
-			case !isSourceEncrypted && !isTargetEncrypted:
-				targetSize, _ = srcInfo.GetActualSize()
-			case isSourceEncrypted && isTargetEncrypted:
-				objInfo := ObjectInfo{Size: actualSize}
-				targetSize = objInfo.EncryptedSize()
-			case !isSourceEncrypted && isTargetEncrypted:
-				targetSize = srcInfo.EncryptedSize()
-			case isSourceEncrypted && !isTargetEncrypted:
-				targetSize, _ = srcInfo.DecryptedSize()
-			}
+			// var targetSize int64
+
+			// switch {
+			// case isDstCompressed:
+			// 	targetSize = -1
+			// case !isSourceEncrypted && !isTargetEncrypted:
+			// 	targetSize, _ = srcInfo.GetActualSize()
+			// case isSourceEncrypted && isTargetEncrypted:
+			// 	objInfo := ObjectInfo{Size: actualSize}
+			// 	targetSize = objInfo.EncryptedSize()
+			// case !isSourceEncrypted && isTargetEncrypted:
+			// 	targetSize = srcInfo.EncryptedSize()
+			// case isSourceEncrypted && !isTargetEncrypted:
+			// 	targetSize, _ = srcInfo.DecryptedSize()
+			// }
```

Comment:
As you can see mint tests have failed, this code exists for a reason @krisis - it is not the only case that matters all the cases here need to be handled separately.

Reply:
@harshavardhana agreed. I shall look into the unit test failures to understand this better.